src/ripple/basics/PerfLog.h

```diff
@@ -0,0 +1,157 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2018 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_BASICS_PERFLOG_H
+#define RIPPLE_BASICS_PERFLOG_H
+
+#include <ripple/protocol/TER.h>
+#include <ripple/core/JobTypes.h>
+#include <ripple/json/json_value.h>
+#include <string>
+#include <cstdint>
+#include <memory>
+#include <type_traits>
+#include <chrono>
+
+namespace ripple {
+namespace perf {
+
+/**
+ * Singleton class that maintains performance counters and optionally
+ * writes Json-formatted data to a distinct log. It should exist prior
+ * to other objects launched by Application to make it accessible for
+ * performance logging.
+ */
+
+class PerfLog
+{
+public:
+    /**
```

Comment:
There are lots uses of `std::chrono::time_point<std::chrono::steady_clock>` in this pull request.  It's a great type, but it's a lot of characters to read.  You can make that easier to read by adding a using declaration at the top here. Consider something like:
```
using steady_time_point = std::chrono::time_point<std::chrono::steady_clock>;
```
Then replace uses of the longer declaration with `steady_time_point` throughout the PerfLog files.

You may want to do something similar with, say, `duration_microseconds`.  But I'll leave that to your judgement.